pandas/core/arrays/boolean.py

```diff
@@ -387,7 +382,7 @@ def __setitem__(self, key, value):
         self._data[key] = value
         self._mask[key] = mask
 
-    def astype(self, dtype, copy=True):
+    def astype(self, dtype, copy: bool = True) -> Union[np.ndarray, BaseMaskedArray]:
```

Comment:
Is there a differentiator between this and `ArrayLike` from pandas._typing (save the TypeVar)?

Reply:
I guess we could, since I assume that astype should eventually support other ExtensionArrays other than IntegerArray and BooleanArray.